iree/compiler/Dialect/HAL/Target/LLVM/LLVMTargetOptions.cpp

```diff
@@ -26,17 +27,34 @@ namespace HAL {
 
 LLVMTargetOptions getDefaultLLVMTargetOptions() {
   LLVMTargetOptions targetOptions;
+
   // Host target triple.
   targetOptions.targetTriple = llvm::sys::getDefaultTargetTriple();
+  targetOptions.targetCPU = llvm::sys::getHostCPUName().str();
+  {
+    llvm::SubtargetFeatures features;
+    llvm::StringMap<bool> hostFeatures;
+    if (llvm::sys::getHostCPUFeatures(hostFeatures)) {
+      for (auto &f : hostFeatures) {
+        features.AddFeature(f.first(), f.second);
+      }
+    }
+    targetOptions.targetCPUFeatures = features.getString();
+  }
+
   // LLVM loop optimization options.
   targetOptions.pipelineTuningOptions.LoopInterleaving = true;
   targetOptions.pipelineTuningOptions.LoopVectorization = true;
   targetOptions.pipelineTuningOptions.LoopUnrolling = true;
+
   // LLVM SLP Auto vectorizer.
   targetOptions.pipelineTuningOptions.SLPVectorization = true;
-  // LLVM -O3.
-  targetOptions.optLevel = llvm::PassBuilder::OptimizationLevel::O3;
+
+  // LLVM -O2.
+  // TODO(benvanik): add an option for this.
+  targetOptions.optLevel = llvm::PassBuilder::OptimizationLevel::O2;
```

Comment:
Can we keep `-O3` as default, or it was too aggressive ?

Reply:
We probably could - it'd be best to add an option for it :)

Reply:
I think I had dropped it down here just to speed up my incremental build times, as until scotts deduping change that just landed (woo!) it would take quite a while to build all 1000 kernels at O3 :) Will make sure it's bumped back up.